Apps/W1/Shopify/test/Catalogs/ShpfyCatalogPricesTest.Codeunit.al

```diff
@@ -72,4 +73,241 @@ codeunit 139646 "Shpfy Catalog Prices Test"
+        ProductInitTest: Codeunit "Shpfy Product Init Test";
+        CatalogInitialize: Codeunit "Shpfy Catalog Initialize";
+        CompanyInitialize: Codeunit "Shpfy Company Initialize";
+        ProductPriceCalculation: Codeunit "Shpfy Product Price Calc.";
+        InitUnitCost: Decimal;
+        InitPrice: Decimal;
+        InitPerc: Decimal;
+        CustDiscPerc: Decimal;
+        UnitCost: Decimal;
+        Price: Decimal;
+        ComparePrice: Decimal;
+    begin
+        // [GIVEN] Setting up shop, catalog, item, and customer-specific pricing.
+        Shop := InitializeTest.CreateShop();
+        CompanyInitialize.CreateShopifyCompany(ShopifyCompany);
+        Catalog := CatalogInitialize.CreateCatalog(ShopifyCompany);
+        CatalogInitialize.CopyParametersFromShop(Catalog, Shop);
+        InitUnitCost := Any.DecimalInRange(10, 100, 1);
+        InitPrice := Any.DecimalInRange(2 * InitUnitCost, 4 * InitUnitCost, 1);
+        CustDiscPerc := Any.DecimalInRange(5, 20, 1);
+        Item := ProductInitTest.CreateItem(Shop."Item Templ. Code", InitUnitCost, InitPrice);
+
+        // [WHEN] Calculating prices without discounts applied.
+        ProductPriceCalculation.SetShopAndCatalog(Shop, Catalog);
+        ProductPriceCalculation.CalcPrice(Item, '', '', UnitCost, Price, ComparePrice);
+
+        // [THEN] Verifying initial prices match expectations.
+        LibraryAssert.AreEqual(InitUnitCost, UnitCost, 'Initial unit cost should match.');
+        LibraryAssert.AreEqual(InitPrice, Price, 'Initial price should match before discount.');
+
+#if CLEAN23
+        // Creating a customer entry, though it is generic as discounts apply to all customers.
+        LibrarySales.CreateCustomer(Customer);
+        // [GIVEN] Applying a universal discount for all customers.
+        ProductInitTest.CreateCustomerPriceList(Shop.Code, Item."No.", InitPrice, CustDiscPerc, Customer);
+        ProductInitTest.CreateAllCustomerPriceList(Shop.Code, Item."No.", InitPrice, InitPerc);
+        Catalog."Customer No." := Customer."No.";
+        Catalog.Modify();
+
+        // [WHEN] Recalculating prices with discounts.
+        ProductPriceCalculation.SetShopAndCatalog(Shop, Catalog);
+        ProductPriceCalculation.CalcPrice(Item, '', '', UnitCost, Price, ComparePrice);
+
+        // [THEN] Confirming discounts are accurately reflected in the final prices.
+        LibraryAssert.AreEqual(InitUnitCost, UnitCost, 'Unit cost should remain consistent.');
+        LibraryAssert.AreEqual(InitPrice, ComparePrice, 'Compare price should reflect initial pricing.');
+        LibraryAssert.AreNearlyEqual(InitPrice * (1 - CustDiscPerc / 100), Price, 0.01, 'Discounted price should be accurately calculated.');
+#endif
+    end;
+
+    [Test]
+    procedure UnitTestCalcCustomerDiscountCatalogPrice()
+    var
+        Shop: Record "Shpfy Shop";
+        Catalog: Record "Shpfy Catalog";
+        LibrarySales: Codeunit "Library - Sales";
```

Comment:
I had to make some more changes to make this pass our build system. This variable for instance is only needed with CLEAN23 tag. But it should all have been taken care of now.

Reply:
Thanks @JesperSchulz